docs/specs/lexicon.md

```diff
@@ -0,0 +1,111 @@
+# Lexicon Schema Documents
+
+Lexicon is a schemas document format used to define [XRPC](./xrpc.md) methods and [ATP Repository](./adx/repo.md) record types. Every Lexicon schema is written in JSON and follows the interface specified below. The schemas are identified using [NSIDs](./nsid.md) which are then used to identify the methods or record types they describe.
+
+## Interface
+
+```typescript
+interface LexiconDoc {
+  lexicon: 1
+  id: string // an NSID
+  type: 'query' | 'procedure' | 'record'
+  revision?: number
```

Comment:
we should have a convo about schema revisions at some point. both how they're handled in the code & the conventions we want to push around them.

imm worried that a `revision` field might encourage people to think about versioning the wrong way

Reply:
Okay sgtm